doc/api/dns.md

```diff
@@ -195,6 +196,15 @@ Uses the DNS protocol to resolve mail exchange records (`MX` records) for the
 contain an array of objects containing both a `priority` and `exchange`
 property (e.g. `[{priority: 10, exchange: 'mx.example.com'}, ...]`).
 
+## dns.resolveNaptr(hostname, callback)
+
+Uses the DNS protocol to resolve regular expression based records (`NAPTR`
+records) for the `hostname`. The `addresses` argument passed to the `callback`
```

Comment:
I really liked your suggestion to emulate `dns.resolveSoa` -- do you think its redundant to have the callback signature and the list of properties?